saleor/app/tasks.py

```diff
@@ -46,3 +51,58 @@ def install_app_task(job_id, activate=False):
         app_installation.message = "Unknown error. Contact with app support."
     app_installation.status = JobStatus.FAILED
     app_installation.save(update_fields=["message", "status"])
+
+
+def _raw_remove_deliveries(deliveries_ids):
+    deliveries = EventDelivery.objects.filter(id__in=deliveries_ids)
+    payloads_ids = list(
+        EventPayload.objects.filter(
+            Exists(deliveries.filter(payload_id=OuterRef("id")))
+        ).values_list("id", flat=True)
+    )
+    payloads = EventPayload.objects.filter(id__in=payloads_ids)
+    attempts = EventDeliveryAttempt.objects.filter(
+        Exists(deliveries.filter(id=OuterRef("delivery_id")))
+    )
+    attempts._raw_delete(attempts.db)  # type: ignore[attr-defined] # raw access # noqa: E501
+    deliveries._raw_delete(deliveries.db)  # type: ignore[attr-defined] # raw access # noqa: E501
+    payloads._raw_delete(payloads.db)  # type: ignore[attr-defined] # raw access # noqa: E501
+
+
+@celeryconf.app.task
+def remove_apps_task():
+    app_delete_period = timezone.now() - settings.DELETE_APP_TTL
+    apps = App.objects.filter(removed_at__lte=app_delete_period)
+
+    # Saleor need to remove app by app to prevents timeous
```

Comment:
Shouldn't there be `timeouts`? 🤔 
```suggestion
    # Saleor needs to remove app by app to prevent timeouts
```